R/derive_param_tte.R

```diff
@@ -845,15 +845,35 @@ print.tte_source <- function(x, ...) {
   }
 }
 
-list_tte_source_objects <- function() {
-  # Get all tte_source objects exported by admiral
-  exported <- getNamespaceExports("admiral")
-  objects <-
-    exported[map_lgl(exported, function(obj_name) {
-      inherits(get(obj_name), "tte_source")
-    })]
 
-  rows <- lapply(objects, function(obj_name) {
+#' List all `tte_source` Objects Available in a Package
+#'
+#' @param package The name of the package in which to search for `tte_source` objects
+#'
+#' @return
+#' A `data.frame` where each row corresponds to one `tte_source` object or `NULL`
+#' if `package` does not contain any `tte_source` objects
+#'
+#' @export
+#'
+#' @examples
+#' list_tte_source_objects()
+list_tte_source_objects <- function(package = "admiral") {
+  assert_character_scalar(package)
+
+  if (!requireNamespace(package, quietly = TRUE)) {
+    err_msg <- sprintf("No package called '%s' is installed and hence no `tte_source` objects are available", package)
+    abort(err_msg)
+  }
+
+  # Get all `tte_source` objects exported by `package`
+  exports <- getNamespaceExports(package)
+  is_tte_source <- map_lgl(exports, function(obj_name) {
+    inherits(get(obj_name), "tte_source")
```

Comment:
If the package is installed but not loaded, this fails.

Reply:
Good catch, I wasn't even aware of that. I exchange `get()` with `getExportedValue()`. That should do the trick.